packages/@webex/plugin-cc/src/AgentProfile/AgentProfile.ts

```diff
@@ -0,0 +1,108 @@
+import {AgentProfileResponse} from './types';
+import AgentProfileService from '../AgentProfileService/AgentProfileService';
+import {
+  DesktopProfileResponse,
+  ListAuxCodesResponse,
+  ListTeamsResponse,
+  UserResponse,
+} from '../AgentProfileService/types';
+import {WebexSDK} from '../types';
+import {attributes, page, pageSize} from '../AgentProfileService/constants';
+
+export default class AgentProfile {
+  ciUserId: string;
+  agentProfile: AgentProfileResponse;
+  webex: WebexSDK;
+  wccAPIURL: string;
+
+  constructor(ciUserId: string, webex: WebexSDK, wccAPIURL: string) {
+    this.webex = webex;
+    this.ciUserId = ciUserId;
+    this.agentProfile = {
+      userDetails: {} as UserResponse['body'],
```

Comment:
If we redesign the interfaces, we can avoid this ['body'] etc. It'll look cleaner.

Reply:
Actually the API data that we are getting is inside the body so to handle that I have initialized this with ['body']. I will see how we can redesign the interface to avoid this.